No tracking Number for return product tab

okom3pom opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | No tracking Number for return product.
| Type?             | bug fix
| Category?         |  BO
| BC breaks?        |  yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/19200
| How to test?      | Made a return and check that the words are right. DevCore must check for deleted variables
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

**:notebook: BC break** 
* Remove property `$carrierId` in `OrderReturnForViewing.php`
* Remove property `$trackingNumber` in `OrderReturnForViewing.php`
* Remove property `$trackingUrlr` in `OrderReturnForViewing.php`

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/26056)
<!-- Reviewable:end -->

Touxten committed:
Good Wording

Touxten committed:
Replace Status name By Status

Touxten committed:
Good Return number

Touxten committed:
Wording

Touxten committed:
Feature : add link

matks commented on src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php:
```diff
@@ -573,11 +573,15 @@ private function getOrderReturns(Order $order): OrderReturnsForViewing
             }
 
+            $conf = $this->configuration->get('PS_RETURN_PREFIX');
+            $trackingNumber = sprintf(
```
Logic seems good in general. But if I understand correctly, actually this is not a tracking number, right ? This is a "merchandise return ID". So we should modify the variable name to make it correct, wdyt ?

okom3pom replied:
Hi @matks I agree with you but : 

After this RP I wanted to open another issue.
Which will correct the variable names and clean up this file :

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Order/QueryResult/OrderReturnForViewing.php

There are: $carrierId; $trackingUrl; $trackingNumber ;
I don't think they belong here, but it's possible that people are using them. And if I delete them, it's a break in the BC.
This fixed PR https://github.com/PrestaShop/PrestaShop/issues/19200 we need to open another one for the other problem.

Progi1984 replied:
@okom3pom You can change the variable name as the scope remains in the method.

PierreRambaud replied:
I think it should be done in the same PR. :thinking:

okom3pom replied:
I would need to know what I am deleting / rename  in[ this file](https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Domain/Order/QueryResult/OrderReturnForViewing.php)? 

_There are: $carrierId; $trackingUrl; $trackingNumber ;
I don't think they belong here, but it's possible that people are using them. And if I delete them, it's a BC break._


@Progi1984 @PierreRambaud @matks ?

PierreRambaud replied:
For me it should be a new variable as it's not related to the tracking number

okom3pom replied:
@PierreRambaud i remove also `$carrierId; $trackingUrl; $trackingNumber ; `

sowbiba commented on src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php:
```diff
@@ -573,11 +573,15 @@ private function getOrderReturns(Order $order): OrderReturnsForViewing
-                $trackingNumber = $orderReturn['tracking_number'];
             }
 
+            $conf = $this->configuration->get('PS_RETURN_PREFIX');
```
I would replace $conf by $returnPrefixByLang to make it clear

sowbiba commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/merchandise_returns.html.twig:
```diff
@@ -40,11 +40,13 @@
               <a href="{{ return.trackingUrl }}">{{ return.trackingNumber }}</a>
             {% elseif return.trackingNumber %}
-              {{ return.trackingNumber }}
+              <a class="d-print-none" href="{{ getAdminLink('AdminReturn', true, {'id_order_return': return.idOrderReturn, updateorder_return : '', 'back': app.request.uri}) }}">
```
```suggestion
              <a class="d-print-none" href="{{ getAdminLink('AdminReturn', true, {'id_order_return': return.idOrderReturn, 'updateorder_return': '', 'back': app.request.uri}) }}">
```

atomiix commented on src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/merchandise_returns.html.twig:
```diff
@@ -40,11 +40,13 @@
               <a href="{{ return.trackingUrl }}">{{ return.trackingNumber }}</a>
             {% elseif return.trackingNumber %}
-              {{ return.trackingNumber }}
+              <a class="d-print-none" href="{{ getAdminLink('AdminReturn', true, {'id_order_return': return.idOrderReturn, 'updateorder_return': '', 'back': app.request.uri}) }}">
```
Why hiding the tracking number in print mode?

PierreRambaud replied:
If it's not a tracking number, it shouldn't be named `trackingNumber` :thinking:

okom3pom replied:
Hello @PierreRambaud i started a discussion with @matks i wait answer.

This fixed PR #19200 we need to open another one for the other problem. ( a lot )

atomiix commented:
@okom3pom Can you solve the conflict, please? :)

okom3pom commented:
Hello @atomiix 

I am waiting for an answer from @matks  and @PierreRambaud 

https://github.com/PrestaShop/PrestaShop/pull/26056#discussion_r745382364

matks commented:
Answer provided by @PierreRambaud  👍  waiting for the new changes

Touxten committed:
Change variable name

Update src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/merchandise_returns.html.twig

Co-authored-by: Ibrahima SOW <sowbiba@hotmail.com>

Fix rebase

Touxten committed:
Remove useless variables

Fix removed int

fix int

okom3pom commented:
Normaly it's good

Touxten committed:
Fix order retrun link

khouloudbelguith commented:
Hi @okom3pom,

I cannot test this PR.
I have an exception in the FO
![image](https://user-images.githubusercontent.com/16067358/146963127-294219a8-6f7e-4bad-b951-78f4eb1f0707.png)
This PR is blocked by this issue: https://github.com/PrestaShop/PrestaShop/issues/26656

Thanks!

okom3pom commented:
I am working on https://github.com/PrestaShop/PrestaShop/issues/26656 it's after a bad rebase

okom3pom commented:
Hello @khouloudbelguith 

Normaly you can test now

Regards

khouloudbelguith commented:
Hi @okom3pom,

As described in the spec:
` The merchandise return tab lists **the date, the type, the carrier (= return status), and the number (= return reference number)**. When the merchandise return is disabled the tab displayed: "no merchandise returned yet".`

The merchandise return tab is still displayed after disabling it.
Steps to reproduce the issue:
1. Go to BO > Customer Service  > Merchandise Returns page & Enable returns
2. Create an order and change status to shipped
3. Go to FO > account page > Order details page & create a return
4. Go to BO > Customer Service  > Merchandise Returns page & disable returns 
![image](https://user-images.githubusercontent.com/16067358/148942178-e43308a3-038b-411c-87ef-c929baec9176.png)
5. Go to BO > Orders > View last order Created > Merchandise returns Tab > still displayed 
![image](https://user-images.githubusercontent.com/16067358/148942266-61048696-c902-4dd1-b5ce-c3ef4344e8e5.png)


Ping @PrestaShop/product-team I need your confirmation.

Thanks!

okom3pom commented:
Hi @khouloudbelguith 

This PR fixes the problems but not the specification, I think it is better to open a new issue for display.

Because it's another problem? 

Regards

khouloudbelguith commented:
Hi @okom3pom,

In fact, the original issue was updated by the product team & this [spec](https://github.com/PrestaShop/prestashop-specs/pull/196) was updated
I need the confirmation of the @PrestaShop/product-team before creating a new issue.

Thank you!

khouloudbelguith commented:
Hi @okom3pom, 

I checked this PR.
It is ok manually :heavy_check_mark: 
I validate this PR. 
![image](https://user-images.githubusercontent.com/16067358/149378217-8bb9d04c-1dd1-42e4-ab58-c65e5051bc69.png)

But automatic tests should be updated
[Mochawesome Report_marchandise_return.pdf](https://github.com/PrestaShop/PrestaShop/files/7864678/Mochawesome.Report_marchandise_return.pdf)
Ping @PrestaShop/qa-automation 
If we merge this PR, nightly will be NOK.

Thanks!

boubkerbribri commented:
@nesrineabdmouleh can you check and fix this please ?

nesrineabdmouleh committed:
Rebase + Fix 06_merchandiseReturnsTab UI test

boubkerbribri commented:
Hello @okom3pom, 
To fix tests, we were missing a rebase and we done it wrongly first.
But Now it's FIXED (with a cherry-pick of your commits and the UI tests fix).
I'm very Sorry for the inconvenience but Can you check please that nothing is missing from your PR please ?

okom3pom commented:
It"s ok !

boubkerbribri commented:
Thanks @okom3pom (and sorry again)
We will wait for reviews to add QA :heavy_check_mark: .

boubkerbribri reviewed (approved):
Validated by @khouloudbelguith previously and UI tests are fixed by @nesrineabdmouleh. So it's QA :heavy_check_mark:

Progi1984 commented:
Thanks @okom3pom @boubkerbribri @nesrineabdmouleh & @khouloudbelguith
